wordpress-mobile/WordPress-iOS
# Introducing PingHub (...and friends)

This is a first basic approach to #6164. I've added the PingHub client and very basic logic:
- Connect when the app launches/log in/foreground
- Disconnect when app goes to background/sign out
- On push, sync the note pushed
- On delete, delete the note deleted

Things I haven't done yet:

- Reconnect when connection drops, reachability
- Coalescing multiple push events into a full sync
- Smarter sync so we don't always sync when notifications appear as long as PingHub is connected

Also, Carthage. Because Cocoapods won't let me have it my way.

We've chatted about this a few times, and I'd like to propose we start using more frameworks but not more repos, and this is my take on it. I started adding a PingHub target, but I also wanted tests, so that was 2 more targets. Which could be fine for now, but I think it might get noisy if we do a few more components. So I'd rather have a separate project. I think this will also help if we want the framework to support being a Cocoapods/Carthage dependency, but I haven't explored that yet.

Cocoapods won't let me have a framework with an embedded framework, so instead of working around Cococapods one more time I decided to include Starscream as a binary with Carthage (actually [Punic](https://github.com/schwa/punic), hence the `punic.yml`).

To test:

- Launch the app, switch to notifications tab
- From a browser with another account, like a post, the like notification should appear immediately.
- From the browser, unlike the post, the like notification should disappear immediately.

Needs review: @jleandroperez (PingHub) @astralbodies (Project setup, Carthage)
